ledger-tool/src/output.rs

```diff
@@ -303,3 +312,238 @@ impl EncodedConfirmedBlockWithEntries {
+                );
+
+                for reward in rewards {
+                    let sign = if reward.lamports < 0 { "-" } else { "" };
+                    println!(
+                        "    {:<44}  {:^15}  {}◎{:<14.9}  ◎{:<18.9}   {}",
+                        reward.pubkey,
+                        if let Some(reward_type) = reward.reward_type {
+                            format!("{reward_type}")
+                        } else {
+                            "-".to_string()
+                        },
+                        sign,
+                        lamports_to_sol(reward.lamports.unsigned_abs()),
+                        lamports_to_sol(reward.post_balance),
+                        reward
+                            .commission
+                            .map(|commission| format!("{commission:>9}%"))
+                            .unwrap_or_else(|| "    -".to_string())
+                    );
+                }
+            }
+        }
+    }
+}
+
+pub fn output_entry(
+    blockstore: &Blockstore,
+    method: &OutputFormat,
+    slot: Slot,
+    entry_index: usize,
+    entry: Entry,
+) {
+    match method {
+        OutputFormat::Display => {
+            println!(
+                "  Entry {} - num_hashes: {}, hash: {}, transactions: {}",
+                entry_index,
+                entry.num_hashes,
+                entry.hash,
+                entry.transactions.len()
+            );
+            for (transactions_index, transaction) in entry.transactions.into_iter().enumerate() {
+                println!("    Transaction {transactions_index}");
+                let tx_signature = transaction.signatures[0];
+                let tx_status_meta = blockstore
+                    .read_transaction_status((tx_signature, slot))
+                    .unwrap_or_else(|err| {
+                        eprintln!(
+                            "Failed to read transaction status for {} at slot {}: {}",
+                            transaction.signatures[0], slot, err
+                        );
+                        None
+                    })
+                    .map(|meta| meta.into());
+
+                solana_cli_output::display::println_transaction(
+                    &transaction,
+                    tx_status_meta.as_ref(),
+                    "      ",
+                    None,
+                    None,
+                );
+            }
+        }
+        OutputFormat::Json => {
+            // Note: transaction status is not output in JSON yet
+            serde_json::to_writer(stdout(), &entry).expect("serialize entry");
+            stdout().write_all(b",\n").expect("newline");
+        }
+        _ => unreachable!(),
+    }
+}
+
+pub fn output_slot(
+    blockstore: &Blockstore,
+    slot: Slot,
+    allow_dead_slots: bool,
+    method: &OutputFormat,
+    verbose_level: u64,
+    all_program_ids: &mut HashMap<Pubkey, u64>,
+) -> std::result::Result<(), String> {
```

Comment:
Fwiw, I've no objection to changing the import definitions to require `fmt::Result` to be made explicit and allow the std Result to use the shorthand. Seems fair for std to be default.

Reply:
> Seems fair for std to be default.

Good callout and I agree with this sentiment; I'll make the change now